### ISSUE TITLE
PO status update

### ISSUE THREAD

**Issue by nakul80** [frappe/erpnext#44016]
### Information about bug

POs entered are received by entering a Purchase invoice. Update stock was checked and quantity and amount recieved was 101% of the PO amt. This was allowed in settings. Still it shows to Receive and Bill and following errors are there.

![Screenshot 2024-11-07 at 6 27 05 PM](https://github.com/user-attachments/assets/3df9e6de-0950-4a23-8286-b9d17cca857e)

### Module

accounts, buying

### Version

ERPNext: v15.40.0 (HEAD)

Frappe Framework: v15.46.0 (HEAD)

India Compliance: v15.13.5 (HEAD)

Frappe Insights: v2.2.6 (HEAD)

### Installation method

FrappeCloud

**Comment by Nihantra-Patel** [comment]
We tested in the latest version but we haven't found the issue. for better understanding, you have to share a screenshot or video of the process.

[scrnli_Tk91BDg6oBr7gO.webm](https://github.com/user-attachments/assets/d959c5b4-0a1b-4e9b-9566-451e5af6dded)

**Comment by nakul80** [comment]
This is the PO
![PO](https://github.com/user-attachments/assets/fa3db184-3707-4d31-8fef-f082d51a1874)


This is the status of the PO
![PO Status](https://github.com/user-attachments/assets/33d7f91c-60db-4147-9076-43a70707090f)

This is the Invoice
![Purchase Invoice](https://github.com/user-attachments/assets/dc0bf7a0-d585-4605-bfe4-67d413ad838e)


Please note that PO has been manually marked as Completed by me. 
Qty and Amt both are higher in the Invoice as against the PO

**Comment by Nihantra-Patel** [comment]
How to manually mark as Completed?

There is no direct option to mark as Completed, but you can close and hold it. If you want to restrict applying the same rate on purchase transactions, go to the Buying Settings and enable the "Maintain Same Rate Throughout the Purchase Cycle" option.

**Comment by nakul80** [comment]
Sorry I meant marked it as Closed.


Regards,
Nakul

**Comment by Nihantra-Patel** [comment]
Purchase Order is closed means the status of the order is updated to "Closed," but it does not affect the quantities or amounts in the system. The system will retain the originally ordered quantities and values, but no further transactions or updates are allowed.

It is typically used to prevent further actions on the order, even if there are remaining items that were not received.

Closing a PO stops further actions from being taken but does not update the quantities or amounts in the system.

I hope this clears your doubts.

**Comment by nakul80** [comment]
My doubt was not regarding Closed. As you can see, the quantity received
and amount billed is more than the PO. Yet PO status shows less than 100%
and there was showing as To Recieve and Bill as the status. I manually
closed it because of that. But status should never have shown less than
100% if billing of qty and amt is more.


Regards,
Nakul

**Comment by Nihantra-Patel** [comment]
I reviewed both of your images. If the purchase invoice is fully paid, there’s no need to manually close the purchase order (PO). When the amount entered in the purchase invoice matches or exceeds the PO amount, the status should automatically set to "Completed." We tested this in 10-12 purchase orders and didn't find any issues. However, a custom client or server script could interfere with the status update.

If you're expecting the "Completed" status after full payment, remember that it should automatically update when you submit the purchase invoice, which then reflects on the PO. I suggest creating a new purchase order, then generating a purchase invoice from it to see if the process works as expected.
